test/qemu/replace.py

```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env python3
+
+# This test copies the default system qemu and patches the binary to have a
+# special string in the help message. We then look for that in the output of
+# the test to make sure we're using the custom qemu. A more realistic use-case
+# for this feature is to use a fork or patched version of the qemu source and
+# rebuild it in your workload (that is just too slow for a unit test like
+# this).
+
+import shutil
+sysQemu = shutil.which("qemu-system-riscv64")
+shutil.copy(sysQemu, "testQemu")
+
+with open("testQemu", "rb") as input_file:
+    content = input_file.read().replace(b"QEMU emulator version", b"HELLO FROM QEMU TEST\n")
```

Comment:
If the intent is to check that a different executable is run, it is not necessary to patch the QEMU binary.  Just make `testQemu` a wrapper script:
```sh
#!/bin/sh
echo "HELLO FROM QEMU TEST"
exec qemu-system-riscv64 "$@"
```

Reply:
OK mister "do it the obvious and easy way", I'll do it your way. My way makes me look clever, which is totally better than being all stable and easy to understand.